### PR TITLE
feat(cli): add extractor witness gen

### PR DESCRIPTION
**Pull request by lonerapier**
- codegen writes circuit param config to `circuits.json`. Overwrites if config with `circuit-name` already present

## Usage:

pabuild:

```
Usage: pabuild <COMMAND>

Commands:
  parser-witness     
  extractor-witness  
  codegen            
  help               Print this message or the help of the given subcommand(s)
```

codegen:
```
Usage: pabuild codegen [OPTIONS] --circuit-name <CIRCUIT_NAME> --input-file <INPUT_FILE> --lockfile <LOCKFILE> <SUBCOMMAND>

Arguments:
  <SUBCOMMAND>  [possible values: json, http]

Options:
      --circuit-name <CIRCUIT_NAME>  Name of the circuit (to be used in circomkit config)
      --input-file <INPUT_FILE>      Path to the JSON/HTTP file
      --lockfile <LOCKFILE>          Path to the lockfile
  -d, --debug                        Optional circuit debug logs
  -h, --help                         Print help
```

witness:
```
Usage: pabuild parser-witness --input-file <INPUT_FILE> --circuit-name <CIRCUIT_NAME> <SUBCOMMAND>

Arguments:
  <SUBCOMMAND>  [possible values: json, http]

Options:
      --input-file <INPUT_FILE>      Path to the JSON file
      --circuit-name <CIRCUIT_NAME>  Name of the circuit (to be used in circomkit config)
  -h, --help                         Print help

==============================================================================

Usage: pabuild extractor-witness --circuit-name <CIRCUIT_NAME> --input-file <INPUT_FILE> --lockfile <LOCKFILE> <SUBCOMMAND>

Arguments:
  <SUBCOMMAND>  [possible values: json, http]

Options:
      --circuit-name <CIRCUIT_NAME>  Name of the circuit (to be used in circomkit config)
      --input-file <INPUT_FILE>      Path to the JSON file
      --lockfile <LOCKFILE>          Path to the lockfile
  -h, --help                         Print help
```

### CODE REVIEW

File: circuits/test/http/codegen.test.ts [hunk 1]
```diff
@@ -41,11 +41,12 @@ interface Response {
...
 
-function executeCodegen(inputFilename: string, outputFilename: string) {
+function executeCodegen(circuitName: string, inputFileName: string, lockfileName: string) {
     return new Promise((resolve, reject) => {
-        const inputPath = join(__dirname, "..", "..", "..", "examples", "http", "lockfile", inputFilename);
+        const inputFilePath = join(__dirname, "..", "..", "..", "examples", "http", inputFileName);
+        const lockfilePath = join(__dirname, "..", "..", "..", "examples", "http", "lockfile", lockfileName);
 
-        const codegen = spawn("cargo", ["run", "http", "--lockfile", inputPath, "--output-filename", outputFilename]);
+        const codegen = spawn("cargo", ["run", "codegen", "http", "--circuit-name", circuitName, "--input-file", inputFilePath, "--lockfile", lockfilePath]);
```
Comment (Autoparallel) [inline circuits/test/http/codegen.test.ts:49]: good call

**Review by Autoparallel — CHANGES_REQUESTED** [review]
Looks great! Can we add/edit docs to explain how to use the CLI more effectively? Best practices/etc.?

File: circuits/test/http/codegen.test.ts (hunk 1 above)
Comment (Autoparallel) [reply circuits/test/http/codegen.test.ts:49]: Just seeing the usage here is helpful for sure

**Comment by Autoparallel** [comment]
The CLI help menus you shared here are great! 

I'm gonna pull locally and mess around more. I might add/edit docs myself to carry this over.

**Comment by lonerapier** [comment]
added updated docs with proof steps for both json and http

**Review by Autoparallel — APPROVED** [review]
Yes!! So good!

**Pull request merged**